ibm/data_source_ibm_is_image.go

```diff
@@ -159,10 +159,14 @@ func imageGet(d *schema.ResourceData, meta interface{}, name, visibility string)
 			break
 		}
 	}
+
 	for _, image := range allrecs {
 		if *image.Name == name {
 			d.SetId(*image.ID)
 			d.Set("status", *image.Status)
+			if *image.Status == "deprecated" {
+				return fmt.Errorf("Given image %s is deprecated", name)
+			}
```

Comment:
We don't need to throw error if an image is deprecated. Just a message or warning should be fine informing the customer that this image is deprecated and soon will be obsolete.

Reply:
Deprecated image is still usable.